Added a date filter to the search page.

vppatel111 opened the pull request.

vppatel111 committed:
Added a date filter to the search page.

vppatel111 committed:
Add an "ALL" option for MLAs and Ridings filters.

Users can now search for ALL MLAs without having to refresh the page.

vppatel111 committed:
Moved the party filter to the SearchWithFilters component.

vppatel111 committed:
Added material-ui/lab to the dependancies.

vppatel111 committed:
Implemented search filter for parties

zdrever commented:
ALRIGHT. I've got good news and bad news. The good news is that with the changes that I've made to the backend api and DatabaseService, data from the server should be easier to deal with. The bad news is that I think it's better to deal with filtering on the front end, which means redoing some of this work (I can do it if you like). Basically, all that we need on the front end is an array that stores ALL data that has been queried from the backend, and then a filtering function that generates a new array that will be displayed to the user. 

I'll need to go back and make changes to the api endpoints that the front end interacts with because they may have changed slightly. Once I've done that, lets merge my changes to the database service and rest api, and then we can work on the filtering (which should be much easier/more flexible because we aren't passing parameters to a stored db procedure).

vppatel111 committed:
Merge branch 'master' into search-date-filter

vppatel111 committed:
Removed the profile summary from the search page.

vppatel111 committed:
Merge branch 'master' into search-date-filter

vppatel111 committed:
[feat] Updated search page to get all mla summaries.

vppatel111 committed:
[feat] Changed search table to filter on prefetched results.

Also fixed Summary model to properly convert from SQL dates to DateTime objects.

vppatel111 committed:
[feat] Updated search filters to now update summary filters for search table.

vppatel111 committed:
[chore] Remove search results from redux store.

vppatel111 committed:
[feat] Updated search filters to use prefetched mla data.

vppatel111 committed:
[feat] Reimplemented filtering with party filters.

vppatel111 committed:
[feat] Re-implemented an ALL option for MLAs and ridings.
